Habinezajanvier/my-brand-api
# #7 Set development environment

#### What does this PR do?
Set development environment
#### Description of Task to be completed?
- Installing nodeJs
- Integrating mongoDb
 -- https://www.digitalocean.com/community/tutorials/how-to-integrate-mongodb-with-your-node-application
- Installing first dependencies
 -- express
 -- mongoose
 -- morgan
 -- Joi
 -- babel
 -- dotenv
#### How should this be manually tested?
- Clone the repo `git clone https://github.com/Habinezajanvier/my-brand-api.git`
- Switch the working directory `cd my-brand-api`
- Checkout the branch `git checkout ch-setup-development-env-#7`
- Install dependencies `npm install`
- Start the server `npm run dev`
> You should see ` ...App is listening on port 3000` and `connected to mongodb` in your console
- Using postman, send a dummy article of only title and body on `http://localhost:3000/articles` end-point using **POST** request
> You should see a successful message and the created article
- On the same endpoint send a **GET** method 
> You should see all article created so far
#### Any background context you want to provide?
- NodeJs should be installed on your local PC
-- Click [here](https://nodejs.org/en/download/) to download nodeJs
- MongoDB should be installed locally on your PC
-- [Install](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-windows/) mongodb for windows
-- [Install](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-os-x/) mongodb for Mac-os
-- [Install](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-ubuntu/) mongodb on ubuntu
#### What are the relevant pivotal tracker stories?
Click [here](https://trello.com/c/kqP6Npbz/7-setting-development-environment) for relevant **Trello** story
